libnetwork/drivers/bridge/setup_ip_tables_linux.go

```diff
@@ -47,7 +47,17 @@ func setupIPChains(config configuration, version iptables.IPVersion) (natChain *
 
 	natChain, err := iptable.NewChain(DockerChain, iptables.Nat, hairpinMode)
 	if err != nil {
-		return nil, nil, nil, nil, fmt.Errorf("failed to create NAT chain %s: %v", DockerChain, err)
+		if version == iptables.IPv4 {
+			return nil, nil, nil, nil, fmt.Errorf("failed to create NAT chain %s: %v", DockerChain, err)
+		}
+		// Log this error, but don't return it - config.EnableIP6Tables is now true by default,
```

Comment:
Can it still be disabled? If not, then it looks like we have a check above that would never have to be checked; https://github.com/moby/moby/blob/f3f20c3a862ff90c7e7b0ee07dabe2176ed0a170/libnetwork/drivers/bridge/setup_ip_tables_linux.go#L40-L42


Given that that check is present, and that there's a config for it, would it make sense to disable that config if we cannot support it?

Reply:
Thinking here; given that we have an option to disable the features, would it make life easier on us to disable that feature if we cannot support it, so that we don't have to perform special handling in other places 🤔

Reply:
Yes, the default has changed to enable ip6tables by default (and to make it non-experimental). But `"ip6tables": false` in daemon config works.

Now the default is to have IPv6 firewall rules, so that only mapped ports are exposed - I think it'd be quite bad to just give up on ip6tables if they don't work. Particularly n the DinD case, where the host is IPv6-enabled but just doesn't have the kernel module loaded, containers would silently end up with no firewall.

So, I think it's better to require that ip6tables works, or is explicitly disabled. If ip6tables is broken and IPv6 isn't used, or IPv6 isn't available on the host, behaviour will be unchanged - IPv4 will just work. With this PR, it's only when trying to use ip6tables that an error will be raised.

In the DinD case - if the host's moby is >=27.0, it'll also have ip6tables enabled and the kernel module will be loaded before the DinD starts. The issue only arises when the host has an old moby (or ip6tables explicitly disabled). That may hit people in the 27.0 release, if they pick up a 27.0-based DinD image and haven't upgraded their host.

But, the change to enabling ip6tables by default is probably going to be significant for more people. Anyone who's using IPv6 and expecting the container to be wide-open with no port mapping will have to change their config (either to disable ip6tables, or to `-p` the ports they want open - and maybe to opt for direct routing / no-NAT by using `-o com.docker.network.bridge.gateway_mode_ipv6=routed` when creating the network).

So, we have some docs work to do, and 27.0's release note will have to highlight these changes.

Reply:
> In the DinD case - if the host's moby is >=27.0, it'll also have ip6tables enabled and the kernel module will be loaded before the DinD starts. The issue only arises when the host has an old moby (or ip6tables explicitly disabled). That may hit people in the 27.0 release, if they pick up a 27.0-based DinD image and haven't upgraded their host.

I can't reproduce this on Debian that's using nftables now, so I must have made a mistake when testing it the other day.

The outer docker starts normally with ip6tables enabled. But, the `ip6_tables` module needed by a dev container isn't loaded.

So, `ip6tables` works on the host, but a `modprobe ip6_tables` is still needed to make it work in the dev container (and, presumably, for any DinD that's using legacy iptables).

I'm wondering if we should try a `modprobe ip6_tables` if we fail to create ip6 chains (or, the `ip link show ip6_tables` trick that @tianon [mentioned on the ticket](https://github.com/moby/moby/issues/47895#issuecomment-2153540823), the dev container at-least doesn't currently have a `modprobe`) ... but I'm not sure if that'd get us into a mess with bad mixes of nftables/xtables.

We may also need to add `ip6_tables` to [`dockerd-rootless-setuptool.sh`](https://github.com/moby/moby/blob/f6d8ac04baae47502c5634636bdbe7bafae3b5d0/contrib/dockerd-rootless-setuptool.sh#L180-L192).